twill-yarn/src/main/hadoop21/org/apache/twill/internal/yarn/Hadoop21YarnAppClient.java

```diff
@@ -197,6 +213,7 @@ public YarnApplicationReport getReport() {
     public void cancel() {
       try {
         yarnClient.killApplication(appId);
+        yarnClient.stop();
```

Comment:
Alternatively, ProcessControllerImpl#getReport can create+start and stop it each time it gets a report.